src/tools/compiletest/src/header.rs

```diff
@@ -378,7 +378,13 @@ impl TestProps {
                     }
 
                     if let Some(flags) = config.parse_name_value_directive(ln, COMPILE_FLAGS) {
-                        self.compile_flags.extend(split_flags(&flags));
+                        let flags = split_flags(&flags);
+                        for flag in &flags {
+                            if flag == "--edition" || flag.starts_with("--edition=") {
+                                panic!("you must use //@edition to configure the edition");
+                            }
+                        }
+                        self.compile_flags.extend(flags);
```

Comment:
Question: hm, how would a test that wants to exercise `--edition` compiler flag itself work?

Reply:
I suppose I wouldn't mind using a `run-make` test for that purpose I suppose.